src/main/groovy/com/readytalk/gradle/plugins/ci/CiInfoPlugin.groovy

```diff
@@ -0,0 +1,174 @@
+      //Applied piecemeal because the CI collector plugin is too simple
+      apply(InfoBrokerPlugin)
+      apply(BasicInfoPlugin)
+      apply(ScmInfoPlugin)
+      apply(InfoJavaPlugin)
+      apply(InfoPropertiesFilePlugin)
+      apply(InfoJarPropertiesFilePlugin)
+      apply(InfoJarManifestPlugin)
+    }
+  }
+
+  private void addCiInfoExtension() {
+    project.extensions.add(EXTENSION_NAME, this.extension)
+  }
+
+  private void setDefaults() {
+    extension.buildNumber = project.plugins.findPlugin('info-ci')?.extension?.buildNumber ?: 'local'
+    extension.branch = gitRepo.branch
+    extension.masterBranch = isMaster(extension.branch)
+    extension.releaseBranch = isReleaseBranch(extension.branch)
+    extension.release = isReleaseTag(extension.branch)
+  }
+
+  private boolean refResolvesToMaster() {
+    gitRepo.getRef('refs/heads/master')?.objectId?.name == gitRepo.resolve('HEAD')?.name
+  }
+
+  private boolean isMaster(String branch) {
+    branch ==~ /master/ || refResolvesToMaster()
+  }
+
+  private boolean isReleaseTag(String branch) {
+    // TODO: Try to resolve HEAD to release tags
+    branch ==~ /v\d+\.\d+\.\d+/
+  }
+
+  private boolean isReleaseBranch(String branch) {
+    // TODO: Try to resolve HEAD to release branch
+    branch ==~ /release_\d+\.\d+\.\d+/
+  }
+
+  private void populateCiInfo(Map env) {
+    if (env.'TRAVIS') {
+      populateTravisInfo(env)
+    }
+    else if (env.'BUILD_NUMBER' && env.'JOB_NAME') {
+      populateJenkinsInfo(env)
+    }
+
+    // Enable manual override for testing, etc.
+    if (env.'CI'?.toBoolean()) {
+      extension.ci = true
+    }
+  }
+
+  private void populateTravisInfo(Map env) {
+    // Reference: http://docs.travis-ci.com/user/ci-environment/#Environment-variables
+    extension.with {
+      ciProvider = 'travis'
+      env.each { k, v ->
+        if (k.startsWith('TRAVIS_')) {
+          def prop = StringUtils.snakeConvert(k.toLowerCase())
+          setProperty(prop, v)
+        }
+      }
+      buildNumber = env.'TRAVIS_BUILD_NUMBER'
+      branch = env.'TRAVIS_BRANCH'
+      ci = true
+      masterBranch = isMaster(branch) && travisPullRequest == 'false'
+      releaseBranch = isReleaseBranch(branch) && travisPullRequest == 'false'
+      release = isReleaseTag(branch)
+    }
+  }
+
+  private void populateJenkinsInfo(Map env) {
+    extension.with {
+      ciProvider = 'jenkins'
+      buildNumber = env.'BUILD_NUMBER'
+      buildId = env.'BUILD_ID'
+      branch = env.'GIT_BRANCH'
+      ci = true
+    }
+  }
+
+  private void mapInfoBrokerFields() {
+    //Map all info fields into broker plugin
+    project.with {
+      plugins.withId('info-broker') { InfoBrokerPlugin broker ->
+        //Override nebula Build-Number, ours is more thorough
+        afterEvaluate {
+          def buildNumber = broker.container.find { it.name == 'Build-Number' }
+          if (buildNumber) broker.container.remove(buildNumber)
```

Comment:
I suspect this is where things are going sideways with the newest version of the nebula-info plugin (0.2.0+). @stormbeta - What do you think?